scheduler/scheduler.go

```diff
@@ -269,26 +291,37 @@ func generateParentHostOptions(pool *evergreen.ContainerPool) cloud.HostOptions
 	}
 }
 
-// FindAvailableParent finds a parent host that can accommodate container,
-// packing on parent that has task with longest expected finish time
-func findAvailableParent(d distro.Distro) (host.Host, error) {
+// getNumContainersOnParents returns a slice of parents and their respective
+// number of current containers currently running in order of longest expected
+// finish time
+func getNumContainersOnParents(d distro.Distro) ([]containersOnParents, error) {
 	allParents, err := host.FindAllRunningParentsByContainerPool(d.ContainerPool)
 	if err != nil {
-		return host.Host{}, errors.Wrap(err, "Could not find running parent hosts")
+		return nil, errors.Wrap(err, "Could not find running parent hosts")
 	}
 
+	numContainersOnParents := make([]containersOnParents, 0)
 	// parents come in sorted order from soonest to latest expected finish time
 	for i := len(allParents) - 1; i >= 0; i-- {
 		parent := allParents[i]
 		currentContainers, err := parent.GetContainers()
 		if err != nil {
-			return host.Host{}, errors.Wrapf(err, "Could not find containers for parent %s", parent.Id)
+			return nil, errors.Wrapf(err, "Could not find containers for parent %s", parent.Id)
 		}
 		if len(currentContainers) < parent.ContainerPoolSettings.MaxContainers {
```

Comment:
Don't you want to add to the return slice regardless of how many containers are on the parent?

Reply:
This isn't that there were no containers found, but that the query to get the containers errored. Instead of returning immediately, maybe I can continue and just not add that parent?